File: src/类型系统/空安全/Person.java
package 类型系统.空安全;

/**
 * @author guang
 * @since 2019-07-27 22:53
 */
public class Person {
    private String name;

    public Person(String name) {
        this.name = name;
    }

    public Person() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }


}

File: src/chapter_4/Person.kt
package chapter_4

class Person(val name: String) {
    // 如果有主构造函数,需要在次构造函数中代理主构造函数
    constructor(name: String, age: Int) : this(name) {
        println("次构造函数:  constructor(name: String, age: Int)")
    }
}

